app/enrichment/create-invoice-number.js

```diff
@@ -1,7 +1,11 @@
 const createInvoiceNumber = (paymentRequest) => {
-  return `S${paymentRequest.invoiceNumber
+  try {
+    return `S${paymentRequest.invoiceNumber
```

Comment:
Do you think splitting these inner code snippets out into sep. vars for readability is valid?

Tests need added to again, trigger the catch clause on this (similar inputs to convert date) but also:

- empty invoice everything else present
- empty invoice + empty agreement + everything else present 
- empty invoice + empty agreement + empty contract number + everything else present
- etc, all perms between the 4 vars (invoice, agreementnumber, contractnumber paymentrequestnumber) I'm sure we can not try if it doesn't have most of them
- also, what is `paymentRequestNumber` surely tht's not the correct key name but rather `paymentRequest`?

Reply:
I think we're going to need to rewrite this full module as part of https://eaflood.atlassian.net/browse/SFI-1971

It was originally written as part of the SFI tech options assessment where we didn't know whether we'd be using Siti Agri or not.  If using Siti Agri we'd get an invoice number and we'd build an invoice from it.  If not using Siti Agri we'd build it from the agreement number.

We need to understand what Vets, Lump Sums and SFI 22 will provide us for invoice numbers and then we can rewrite this to handle all schemes properly.  As we've already got a ticket specifically for this, I think it's ok not to address it here.

`paymentRequestNumber` is the version of the payment request.  ie first payment = `1`, first adjustment = `2`, second adjustment = `3` etc.  So it is the correct property.